### PR TITLE
Add start_on_boot logic

### PR DESCRIPTION
**Pull request by AwaitFuture**
Problem:
* Configuration parameter for start_on_boot doesnt do anything

Solution:
* implement logic

### CODE REVIEW

File: lib/slurp/application.ex [hunk 1]
```diff
@@ -20,5 +20,17 @@ defmodule Slurp.Application do
     Enum.each(new_head_subscriptions, &Slurp.NewHeads.Subscriptions.put/1)
     {:ok, log_subscriptions} = Slurp.Logs.Subscriptions.from_config(blockchains)
     Enum.each(log_subscriptions, &Slurp.Logs.Subscriptions.put/1)
+
+    with b <- blockchains
```
Comment (rupurt) [inline lib/slurp/application.ex:24]: Can you add a new `start_phase` for this as it's not hydration. Something like `:blockchains_and_subscriptions`?

How do you feel about naming the `b` variable something like `ids`?

File: lib/slurp/application.ex [hunk 2]
```diff
@@ -20,5 +20,17 @@ defmodule Slurp.Application do
...
+      |> Enumerati.filter([start_on_boot: true])
+      |> Enum.map(fn m -> Map.get(m, :id) end)
```
Comment (rupurt) [inline lib/slurp/application.ex:26]: This should use struct attribute access. Also how do you feel about using the anonymous function shorthand? e.g.

```elixir
|> Enum.map(& &1.id)
```

**Review by rupurt — CHANGES_REQUESTED** [review]
Nice one! Just added a couple of minor comments

File: lib/slurp/application.ex (hunk 2 above)
Comment (AwaitFuture) [reply lib/slurp/application.ex:26]: Im fine with the shorthand here, will swap.

File: lib/slurp/application.ex (hunk 1 above)
Comment (AwaitFuture) [reply lib/slurp/application.ex:24]: Sure, Ill add a new start phase.

**Review by rupurt — APPROVED** [review]

**Comment by rupurt** [comment]
:shipit:

**Pull request merged**